packages/react/tooltip/src/Tooltip.tsx

```diff
@@ -61,109 +128,81 @@ interface TooltipProps {
   const contentId = useId();
+  const openTimerRef = React.useRef(0);
+  const delayDuration = delayDurationProp ?? context.delayDuration;
+  const openDelay = context.isOpenDelayed ? delayDuration : 0;
+  const wasOpenDelayed = usePrevious(Boolean(openDelay));
   const [open = false, setOpen] = useControllableState({
     prop: openProp,
     defaultProp: defaultOpen,
     onChange: onOpenChange,
   });
-  const [stateAttribute, setStateAttribute] = React.useState<StateAttribute>(
-    openProp ? 'instant-open' : 'closed'
-  );
 
-  // control open state using state machine subscription
-  React.useEffect(() => {
-    const unsubscribe = stateMachine.subscribe(({ state, context }) => {
-      if (state === 'open' && context.id === contentId) {
-        setOpen(true);
-      } else {
-        setOpen(false);
-      }
-    });
-
-    return unsubscribe;
-  }, [contentId, setOpen]);
-
-  // sync state attribute with using state machine subscription
+  const stateAttribute = React.useMemo(() => {
+    return open ? (wasOpenDelayed ? 'delayed-open' : 'instant-open') : 'closed';
+  }, [wasOpenDelayed, open]);
+
+  const prevOpen = usePrevious(open);
+  const { onTooltipOpen, onTooltipClose } = context;
   React.useEffect(() => {
-    const unsubscribe = stateMachine.subscribe(({ state, context }) => {
-      if (context.id === contentId) {
-        if (state === 'open') {
-          setStateAttribute(context.delayed ? 'delayed-open' : 'instant-open');
-        } else {
-          setStateAttribute('closed');
-        }
-      } else {
-        setStateAttribute('closed');
-      }
-    });
-
-    return unsubscribe;
-  }, [contentId]);
-
-  const handleFocus = React.useCallback(
-    () => stateMachine.send({ type: 'FOCUS', id: contentId }),
-    [contentId]
-  );
-  const handleOpen = React.useCallback(
-    () => stateMachine.send({ type: 'OPEN', id: contentId, delayDuration }),
-    [contentId, delayDuration]
-  );
-  const handleClose = useCallbackRef(() => {
-    stateMachine.send({ type: 'CLOSE', id: contentId, skipDelayDuration });
-  });
+    if (prevOpen === open) return;
+    // We do this in an effect to ensure that we don't broadcast an
+    // open/close unless open state change was allowed (when controlled).
+    if (open) onTooltipOpen();
+    else onTooltipClose();
+  }, [prevOpen, open, onTooltipOpen, onTooltipClose]);
 
-  // send transition if the component unmounts
-  React.useEffect(() => () => handleClose(), [handleClose]);
+  React.useEffect(() => {
+    return () => window.clearTimeout(openTimerRef.current);
+  }, []);
 
-  // if we're controlling the component
-  // put the state machine in the appropriate state
-  useLayoutEffect(() => {
-    if (openProp === true) {
-      stateMachine.send({ type: 'OPEN', id: contentId });
-    }
-  }, [contentId, openProp]);
+  const handleOpen = React.useCallback(() => {
+    document.dispatchEvent(new CustomEvent(TOOLTIP_OPEN));
```

Comment:
I dispatch here instead of within the `TooltipProvider` so that tooltips will still close if another tooltip is opened when there isn't a `TooltipProvider`.